pkg/resource/archive/hooks.go

```diff
@@ -0,0 +1,99 @@
+package archive
+
+import (
+	"errors"
+	"fmt"
+
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+
+	"github.com/aws-controllers-k8s/eventbridge-controller/apis/v1alpha1"
+)
+
+const (
+	StatusEnabled      = "ENABLED"
+	StatusDisabled     = "DISABLED"
+	StatusCreating     = "CREATING"
+	StatusUpdating     = "UPDATING"
+	StatusCreateFailed = "CREATE_FAILED"
+	StatusUpdateFailed = "UPDATE_FAILED"
+)
```

Comment:
No need to repeat these constants. You can use the existing [`svcsdk.ArchiveStateEnabled` through `svcsdk.ArchiveStateUpdateFailed`](https://github.com/aws/aws-sdk-go/blob/1c759fb5b09ba21a4a82883f8a2a1a411c814ad4/service/eventbridge/api.go#L17706-L17722) constants.